crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -879,9 +901,96 @@ impl HolochainP2pActor {
             kitsune,
             pending,
             request_duration_metric: create_p2p_request_duration_metric(),
+            pruning_task_abort_handle,
         }))
     }
 
+    // Prunes expired URLs at an interval and checks the peer store for agent infos of unresponsive
+    // URLs. If there is an updated agent info since the URL was marked unresponsive, the URL will
+    // be pruned.
+    fn spawn_pruning_task(
+        interval_ms: u64,
+        kitsune2: DynKitsune,
+        db_getter: GetDbPeerMeta,
+    ) -> AbortHandle {
+        tokio::spawn(async move {
+            loop {
+                tokio::time::sleep(Duration::from_millis(interval_ms)).await;
+
+                let spaces = kitsune2.list_spaces();
+                let pruning_futs =
+                    spaces.into_iter().map(|space_id| {
+                        let db_getter = db_getter.clone();
+                        let kitsune2 = kitsune2.clone();
+                        async move {
+                            let space = kitsune2.clone().space(space_id.clone()).await?;
+                            let peer_store = space.peer_store().clone();
+                            let db = db_getter(DnaHash::from_k2_space(&space_id)).await?;
+                            // Prune any expired entries.
+                            db.write_async(|txn| -> DatabaseResult<()> {
+                                let prune_count = txn.execute(sql_peer_meta_store::PRUNE, [])?;
```

Comment:
I see that the tests pass but I'm confused by the logic. When I look at the SQL query in `prune.sql` it looks like it removes anything where `expires_at < unixepoch() * 1000000`. How does that remove anything ever?

Reply:
Ah, I misinterpreted what `unixepoch()` returns. I thought it returns the timestamp of back in 1970 but it returns the current time in unix epoch time.

Reply:
Yes, I learned that too. It's the current time since unix epoch, in seconds.